code/modules/client/verbs/ooc.dm

```diff
@@ -259,9 +259,19 @@ GLOBAL_VAR_INIT(normal_ooc_colour, OOC_COLOR)
 	set category = "OOC"
 	set desc ="Ignore a player's messages on the OOC channel"
 
-	var/selection = input("Please, select a player!", "Ignore", null, null) as null|anything in sortKey(GLOB.clients)
-	if(!selection)
+	
+	var/see_ghost_names = isobserver(mob)
+	var/list/choices = list()
+	for(var/client/C in GLOB.clients)
+		if(isobserver(C.mob) && see_ghost_names)
+			choices["[C.mob]([C])"] = C
+		else
+			choices["[C]"] = C
```

Comment:
Is stringifying necessary here?

Reply:
Technically no, i'll drop it in a moment.